src/components/NewQuestion.jsx

```diff
@@ -0,0 +1,293 @@
+  Label,
+  Input,
+  FormText,
+  FormFeedback,
+} from 'reactstrap'
+import { useInput } from 'react-hanger'
+import classNames from 'classnames'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faChevronDown } from '@fortawesome/free-solid-svg-icons'
+import getConfig from 'next/config'
+
+import constants from '../constants'
+import { CREATE_QUESTION } from '../constants/ActionTypes'
+import UserAutocomplete from './UserAutocomplete'
+
+const { uidName, uidArticle } = getConfig().publicRuntimeConfig
+
+const validateField = (value, maxLength) => {
+  if (!value) {
+    return 'This field is required!'
+  }
+  if (value.length > maxLength) {
+    return `This field has a maximum length of ${maxLength} characters`
+  }
+  return undefined
+}
+
+const hasErrors = invalidFields => {
+  return Object.keys(invalidFields).reduce((isInvalid, fieldName) => {
+    return isInvalid || !!invalidFields[fieldName]
+  }, false)
+}
+
+const isInvalid = error => error !== undefined && error !== ''
+
+const capitalizeString = string =>
+  string.charAt(0).toUpperCase() + string.slice(1)
+
+const NewQuestion = props => {
+  const [isOpen, setIsOpen] = useState(!props.isUserCourseStaff)
+  const [pendingUser, setPendingUser] = useState([])
+  const nameInput = useInput(
+    props.isUserCourseStaff ? '' : props.user.name || ''
+  )
+  const topicInput = useInput('')
+  const locationInput = useInput('')
+  const [submitInProgress, setSubmitInProgress] = useState(false)
+  const [fieldErrors, setFieldErrors] = useState({})
+
+  const onCardHeaderClick = () => {
+    if (!props.isUserCourseStaff) {
+      // This isn't toggleable for normal users
+      return
+    }
+    setIsOpen(current => !current)
+  }
+
+  const onUserChanged = users => {
+    setPendingUser(users)
+    if (users.length === 1 && users[0].name && !nameInput.value) {
+      nameInput.setValue(users[0].name)
+    }
+  }
+
+  const onSubmit = e => {
+    e.preventDefault()
+    const newFieldErrors = {}
+    // Skip location validation if queue is in fixed location mode
+    if (!props.queue.fixedLocation) {
+      newFieldErrors.location = validateField(
+        locationInput.value,
+        constants.QUESTION_LOCATION_MAX_LENGTH
+      )
+    }
+    newFieldErrors.name = validateField(
+      nameInput.value,
+      constants.QUESTION_NAME_MAX_LENGTH
+    )
+    newFieldErrors.topic = validateField(
+      topicInput.value,
+      constants.QUESTION_TOPIC_MAX_LENGTH
+    )
+    setFieldErrors(newFieldErrors)
+    if (hasErrors(newFieldErrors)) return
+
+    const uid = props.isUserCourseStaff
+      ? (pendingUser[0] && pendingUser[0].uid) || undefined
+      : undefined
+    const question = {
+      uid,
+      name: nameInput.value,
+      location: locationInput.value,
+      topic: topicInput.value,
+    }
+    // Disable the button while the request is in flight
+    setSubmitInProgress(true)
+    props.createQuestion(props.queueId, question).then(action => {
```

Comment:
Could we also re-enable the button in case the request fails for whatever reason?